src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -22,14 +22,14 @@ public class Tag {
     public Tag(String tagName) {
         requireNonNull(tagName);
         checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
-        this.tagName = tagName;
+        this.tagName = new Name(tagName);
     }
 
     /**
      * Returns true if a given string is a valid tag name.
      */
     public static boolean isValidTagName(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return test.length() <= 30;
```

Comment:
I think 30 should be a private static final field rather than over here. Also need to check if the length is more than 0. i.e. "" is not a valid tag name.

Reply:
Good catch, will update accordingly.